rspec-mode.el

```diff
@@ -718,8 +729,21 @@ or a cons (FILE . LINE), to run one example."
 (define-compilation-mode rspec-compilation-mode "RSpec Compilation"
   "Compilation mode for RSpec output."
   (add-hook 'compilation-filter-hook 'rspec-colorize-compilation-buffer nil t)
+  (add-hook 'compilation-finish-functions 'rspec-store-failures nil t)
   (add-hook 'compilation-finish-functions 'rspec-handle-error nil t))
 
+(defun rspec-store-failures (&rest ignore)
+  "Store the file and line number of the failed examples from this run."
+  (let* ((body (buffer-string))
+          (failure-regexp "^rspec \\([0-9A-Za-z@_./:-]+\\.rb:[0-9]+\\)")
+          (failures (save-match-data
+                      (let ((pos 0) matches)
+                        (while (string-match failure-regexp body pos)
```

Comment:
This would be better performed with `re-search-forward`. You don't copy the buffer text into a string; instead use `save-excursion`, `(goto-char (point-min))`, and traverse the buffer till the end while saving the matches,